x-pack/legacy/plugins/apm/server/lib/apm_telemetry/index.ts

```diff
@@ -4,9 +4,57 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export {
-  storeApmTelemetry,
-  createApmTelementry,
-  APM_TELEMETRY_DOC_ID
-} from './apm_telemetry';
-export { makeApmUsageCollector } from './make_apm_usage_collector';
+import { Server } from 'hapi';
+import { countBy } from 'lodash';
+import { SavedObjectAttributes } from 'src/core/server';
+import { InternalCoreSetup } from 'src/core/server';
+import { isAgentName } from '../../../common/agent_name';
+import { getSavedObjectsClient } from '../helpers/saved_objects_client';
+export const APM_TELEMETRY_DOC_ID = 'apm-telemetry';
+
+export function createApmTelementry(
+  agentNames: string[] = []
+): SavedObjectAttributes {
+  const validAgentNames = agentNames.filter(isAgentName);
+  return {
+    has_any_services: validAgentNames.length > 0,
+    services_per_agent: countBy(validAgentNames)
+  };
+}
+
+export async function storeApmTelemetry(
+  server: Server,
+  apmTelemetry: SavedObjectAttributes
+) {
+  try {
+    const savedObjectsClient = getSavedObjectsClient(server);
+    await savedObjectsClient.create('apm-telemetry', apmTelemetry, {
```

Comment:
Shouldn't this use the constant `APM_TELEMETRY_DOC_ID`?

Reply:
I didn't do this because it would conflate the saved object type (the string literal) with the saved object id (the constant). I could alternatively make a second constant `APM_TELEMETRY_SAVED_OBJ_TYPE`  or rename `APM_TELEMETRY_DOC_ID` to `APM_TELEMETRY_DOC_ID_AND_SAVED_OBJ_TYPE` and use it in both places.

Reply:
I think having two constants makes sense. You did something similar of ui indices: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/server/lib/settings/apm_indices/get_apm_indices.ts#L26-L27

Reply:
I'm wondering if it's better to have a `apm_saved_object_constants.ts` file where all saved objects are defined?